### ISSUE TITLE
Display minicart message if cart is empty

### ISSUE THREAD

**Issue by nattywebdev** [backdrop-contrib/uc_minicart#1]
At present this neat module displays the minicart if there is something present but is not displayed if the cart is empty.
Is it possible to display a message in its place, such as 'Cart is empty'?

**Comment by nattywebdev** [comment]
I've cloned this module as uc_minibasket for my own purposes, to add the functionality mentioned earlier in May (above).
I also wanted to display, to meet customer requirements and to make the display very tight:
![image](https://github.com/user-attachments/assets/89ef70c4-7516-4558-b775-dd3ad5c5b0d2)

That is:
1. a basket instead of a cart icon;
2. if the basket (cart) is empty show the icon and add a title (basket is empty) and have no link added to the icon;
3. a small number shown next to the basket icon to indicate number;
4. omit the checkout icon and link;

I've managed all of these except for removing the link to cart when the cart is empty.

Where in the module does this happen?

I assumed it was built here:
`        $summary_links = array(
          'cart-block-view-cart' => array(
            'title' => t('View cart'),
            'href' => 'cart',
            'attributes' => array('rel' => 'nofollow'),
          ),
        );`

but that's not the whole story, as there's also this line:
`$block_content .= l('<i class="fa-solid fa-basket-shopping" title="' . t('Basket empty') . '"></i>' , 'cart', array('html' => TRUE));`

I'd assumed the 'cart' in that line was providing the link and I'm almost there - but replacing 'cart' with '' simply gives a cart icon with a link to nowhere (or rather, <front>).

@laryn could you possibly comment for me on this?

**Comment by laryn** [comment]
@nattywebdev Are you interested in co-maintaining this module and getting your improvements in here? I filed another issue related to icons, and it could potentially be expanded to allow selection of cart or basket, perhaps. https://github.com/backdrop-contrib/uc_minicart/issues/2

The issue you are running into here is that the entire thing is being run through `l()` which has the sole purpose of creating a link. So in the case of an empty cart, you'd just need:

`$block_content .= '<i class="fa-solid fa-basket-shopping" title="' . t('Basket empty') . '"></i>';`

**Comment by nattywebdev** [comment]
@laryn I would be happy to do so, as long as you realise I'm a novice when it comes to module coding. As a fairly small module, this would be a good training ground, if you agree (and can help me)!

If so, I'd like to see a couple of options introduced, that I've added to my cloned custom module.

- option to display basket or cart
- option for the standard display (as now) or my new minimalist display, as shown in the message above, that is:
  - basket/cart icon plus a simple number showing basket content quantity 
  - omit text showing quantity and value
  - omit checkout icon

If you agree then I'm up for it - otherwise, I'll keep my changes as a separate module.

**Comment by laryn** [comment]
@nattywebdev Yes, sure, I'm up for this as time allows. Let's start off by having you submit PRs on various tickets and I'll try to review them in a timely manner. Once you're getting the hang of things we can make you an official co-maintainer of this module.

Does this plan sound good? Can you open individual tickets for each feature change, so we can work on these individually?

**Comment by nattywebdev** [comment]
Ok, will do.

**Comment by nattywebdev** [comment]
@laryn I've just tried to apply the latest update (from 1.x-1.0.0 to  1.x-1.1.1) and am unable to apply it. I assume this is because I cloned the module as uc_minibasket and there is some conflict. Can you advise please?

What happens is the Available Updates report shows that 1.x-1.1.1 is available, so I apply it and the files download successfully). But the report still says that the update is available, as you can see here:

![Image](https://github.com/user-attachments/assets/b88b6270-46f4-427f-8aee-0fb6bd6ab93a)

Any help you can suggest will be most welcome!

**Comment by nattywebdev** [comment]
Update: I tried the same on my staging copy and see an error message displayed:

![Image](https://github.com/user-attachments/assets/321008a5-a912-4367-b522-1425db6f003c)

Is this just me that this affects?

**Comment by laryn** [comment]
@nattywebdev I suspect you're right, that it's something about your particular site or setup. I was able to download the module via the zip file successfully and install on a new demo site.

You may wish to detach your modified version from this one so they aren't entangled anymore. You'd still probably want to monitor this module periodically in case of security updates, etc. which you'd want to patch into yours.
